code/game/machinery/announcement_system.dm

```diff
@@ -21,23 +21,55 @@ GLOBAL_LIST_EMPTY(announcement_systems)
 	var/arrivalToggle = 1
 	var/newhead = "%PERSON, %RANK, is the department head."
 	var/newheadToggle = 1
+	var/factionleave = "%PERSON, the %RANK, has left these lands for a (hopefully) better land in the wasteland." //this is used to announce a faction of when a player leaves by the matrix
+	var/factionleavedead = "the corpse of %PERSON, the %RANK, is now feeding the vultures, at the other side of the border" //same but dead
+
 
 	var/greenlight = "Light_Green"
 	var/pinklight = "Light_Pink"
 	var/errorlight = "Error_Red"
 
+/obj/machinery/announcement_system/Mr_Yuma //ATOM EDIT -Start- , adds a lore friendly announcer in the form of Mr. Yuma, a parody of Mr. New Vegas
```

Comment:
set `arrivalToggle = FALSE` as right now it's announcing everyone arriving over common.

Reply:
rightto, done